### PR TITLE
PanelChrome: Use labelledby for accessible title

### PR DESCRIPTION
**Pull request by tskarhed**
**What is this feature?**

Makes the heading the accessible title.

**Why do we need this feature?**

Using a screen reader without this causes the landmark rotor list to read the title AND all the content, which is quite annoying.

Unfortunately I failed to address this when updating the heading level for PanelChrome.

**Who is this feature for?**

Screen reader users.

**Which issue(s) does this PR fix?**:

Point 2 in #79627

**Special notes for your reviewer:**

Review by listing landmarks with a screen reader before and after the pr.

I am also open for suggestions on how to handle the cases when there is no panel title (it will just default to reading all content)

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by joshhunt** [comment]
This one can be for @grafana/dashboards-squad to review

File: packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx [hunk 1]
```diff
@@ -249,6 +256,7 @@ export function PanelChrome({
     <section
       className={cx(styles.container, { [styles.transparentContainer]: isPanelTransparent })}
       style={containerStyles}
+      aria-labelledby={panelTitleId}
```
Comment (tskarhed) [inline packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx:259]: @ckbedwell This is allowed for `landmark` roles, so have no idea why pa11y is complaining 🤔 Any thoughts?

Comment (tskarhed) [reply packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx:259]: Ah, makes sense 🤦 Thank you for clarifying. "No element with that id" would've been a better error message than "It's invalid"

Comment (ckbedwell) [reply packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx:259]: Tell me about it 😂  It took me a while to work out what the problem was.

**Comment by diegommm** [comment]
Hi! Greetings from the Release Guild. Just letting you know I'm updating your PR's milestone to [11.2.x](https://github.com/grafana/grafana/milestone/529). If you need to see your changes in a previous version, please consider adding a backport label.

Thank you!

**Review by dprokop — APPROVED** [review]

**Pull request merged**